### DISCUSSION TITLE
linux error: corrupted double-linked list

### DISCUSSION THREAD

**Discussion by ESH-y** [firemodels/fds#13582] [Q&A]
Hi,
I run an FDS file on Linux (Revision: FDS-6.9.1-0-g889da6a-release)  and it crashed in the 300 second area. This is the error I got:

corrupted double-linked list

===================================================================================
=   BAD TERMINATION OF ONE OF YOUR APPLICATION PROCESSES
=   RANK 0 PID 471613 RUNNING AT RedHat-8.1
=   KILLED BY SIGNAL: 9 (Killed)
===================================================================================

===================================================================================
=   BAD TERMINATION OF ONE OF YOUR APPLICATION PROCESSES
=   RANK 1 PID 471614 RUNNING AT RedHat-8.1
=   KILLED BY SIGNAL: 9 (Killed)
===================================================================================

===================================================================================
=   BAD TERMINATION OF ONE OF YOUR APPLICATION PROCESSES
=   RANK 2 PID 471615 RUNNING AT RedHat-8.1
=   KILLED BY SIGNAL: 9 (Killed)
===================================================================================

===================================================================================
=   BAD TERMINATION OF ONE OF YOUR APPLICATION PROCESSES
=   RANK 3 PID 471616 RUNNING AT RedHat-8.1
=   KILLED BY SIGNAL: 9 (Killed)
===================================================================================

===================================================================================
=   BAD TERMINATION OF ONE OF YOUR APPLICATION PROCESSES
=   RANK 4 PID 471617 RUNNING AT RedHat-8.1
=   KILLED BY SIGNAL: 9 (Killed)
===================================================================================

===================================================================================
=   BAD TERMINATION OF ONE OF YOUR APPLICATION PROCESSES
=   RANK 5 PID 471618 RUNNING AT RedHat-8.1
=   KILLED BY SIGNAL: 9 (Killed)
===================================================================================

===================================================================================
=   BAD TERMINATION OF ONE OF YOUR APPLICATION PROCESSES
=   RANK 6 PID 471619 RUNNING AT RedHat-8.1
=   KILLED BY SIGNAL: 9 (Killed)
===================================================================================

===================================================================================
=   BAD TERMINATION OF ONE OF YOUR APPLICATION PROCESSES
=   RANK 7 PID 471620 RUNNING AT RedHat-8.1
=   KILLED BY SIGNAL: 6 (Aborted)
===================================================================================

===================================================================================
=   BAD TERMINATION OF ONE OF YOUR APPLICATION PROCESSES
=   RANK 8 PID 471621 RUNNING AT RedHat-8.1
=   KILLED BY SIGNAL: 9 (Killed)
===================================================================================
[jfpe@RedHat-8 006]$ 

Then I tried to restart it and got the following error:

corrupted size vs. prev_size
corrupted size vs. prev_size
[TRUCK_001.fds.txt](https://github.com/user-attachments/files/17381757/TRUCK_001.fds.txt)
[TRUCK_001.fds.txt](https://github.com/user-attachments/files/17381766/TRUCK_001.fds.txt)

ERROR: MPI exchange of particle sizes timed out for MPI process 8 running on RedHat-8.1. FDS will abort.

Abort(1) on node 8 (rank 8 in comm 0): application called MPI_Abort(MPI_COMM_WORLD, 1) - process 8
[jfpe@RedHat-8 restart]$ 

I would appreciate help, please understand what these errors mean? And how can this be fixed?

Attached is the file I run (without the geometry of the project).

**Comment by mcgratta** [comment]
Does this error occur consistently? Have you run jobs like this successfully before?

**Comment by ESH-y** [comment]
hi,
yes it does.
i tried running the same file on another computer (not linox) and i got this error:
forrtl: severe (157): Program Exception - access violation
Image              PC                Routine            Line        Source
fds_openmp.exe     00007FF7A8509E3F  Unknown               Unknown  Unknown
fds_openmp.exe     00007FF7AF912086  Unknown               Unknown  Unknown
fds_openmp.exe     00007FF7AF74C1A2  Unknown               Unknown  Unknown
fds_openmp.exe     00007FF7AF74BAF0  Unknown               Unknown  Unknown
fds_openmp.exe     00007FF7AF8424A9  Unknown               Unknown  Unknown
fds_openmp.exe     00007FF7AF5878B9  Unknown               Unknown  Unknown
fds_openmp.exe     00007FF7AF545DCB  Unknown               Unknown  Unknown
fds_openmp.exe     00007FF7AFEFEBB0  Unknown               Unknown  Unknown
KERNEL32.DLL       00007FFEB49B7974  Unknown               Unknown  Unknown
ntdll.dll          00007FFEB5D4A2F1  Unknown               Unknown  Unknown

===================================================================================
=   BAD TERMINATION OF ONE OF YOUR APPLICATION PROCESSES
=   RANK 0 PID 10348 RUNNING AT NO24
=   EXIT STATUS: -1 (ffffffff)
===================================================================================

===================================================================================
=   BAD TERMINATION OF ONE OF YOUR APPLICATION PROCESSES
=   RANK 1 PID 11844 RUNNING AT NO24
=   EXIT STATUS: -1 (ffffffff)
===================================================================================

===================================================================================
=   BAD TERMINATION OF ONE OF YOUR APPLICATION PROCESSES
=   RANK 2 PID 16380 RUNNING AT NO24
=   EXIT STATUS: -1 (ffffffff)
===================================================================================

===================================================================================
=   BAD TERMINATION OF ONE OF YOUR APPLICATION PROCESSES
=   RANK 3 PID 8272 RUNNING AT NO24
=   EXIT STATUS: -1 (ffffffff)
===================================================================================

===================================================================================
=   BAD TERMINATION OF ONE OF YOUR APPLICATION PROCESSES
=   RANK 4 PID 17272 RUNNING AT NO24
=   EXIT STATUS: -1 (ffffffff)
===================================================================================

===================================================================================
=   BAD TERMINATION OF ONE OF YOUR APPLICATION PROCESSES
=   RANK 5 PID 16288 RUNNING AT NO24
=   EXIT STATUS: -1 (ffffffff)
===================================================================================

===================================================================================
=   BAD TERMINATION OF ONE OF YOUR APPLICATION PROCESSES
=   RANK 6 PID 10688 RUNNING AT NO24
=   EXIT STATUS: -1 (ffffffff)
===================================================================================

===================================================================================
=   BAD TERMINATION OF ONE OF YOUR APPLICATION PROCESSES
=   RANK 8 PID 15700 RUNNING AT NO24
=   EXIT STATUS: -1 (ffffffff)
===================================================================================


can you please helpme to understand what is the error? so that i will know how to fix it.

thank you

**Comment by mcgratta** [comment]
Explain to me how you are attempting to run the job.

**Reply by ESH-y** [reply]
hi, 
this is the line im runnig:
fds_local -p 9 -o 2 TRUCK_001.fds

is it ok?

**Reply by mcgratta** [reply]
No. `fds_local` is a command designed for Windows. It does not work on linux. Read the User's Guide on how to run on linux.

**Reply by ESH-y** [reply]
yes i know.

i tried running o n linox and windows:

1. this is the line i use in windows: fds_local -p 9 -o 2 TRUCK_001.fds and the error i got is: forrtl: severe (157): Program Exception - access violation
Image PC Routine Line Source
fds_openmp.exe 00007FF7A8509E3F Unknown Unknown Unknown
fds_openmp.exe 00007FF7AF912086 Unknown Unknown Unknown
fds_openmp.exe 00007FF7AF74C1A2 Unknown Unknown Unknown
fds_openmp.exe 00007FF7AF74BAF0 Unknown Unknown Unknown
fds_openmp.exe 00007FF7AF8424A9 Unknown Unknown Unknown
fds_openmp.exe 00007FF7AF5878B9 Unknown Unknown Unknown
fds_openmp.exe 00007FF7AF545DCB Unknown Unknown Unknown
fds_openmp.exe 00007FF7AFEFEBB0 Unknown Unknown Unknown
KERNEL32.DLL 00007FFEB49B7974 Unknown Unknown Unknown
ntdll.dll 00007FFEB5D4A2F1 Unknown Unknown Unknown

2. this is the line i use in linox: mpiexec -np 9 fds TRUCK_001.fds and the error i got is: corrupted double-linked list

can you please explain what does it mean this errors i got?

thank you

**Reply by mcgratta** [reply]
This could mean almost anything. I suggest that you create a small case with two or three meshes and a small number of cells each. Maybe just this
```
&HEAD CHID='test' /
&TIME T_END=10 /
&MESH XB=0,1,0,1,0,1, IJK=10,10,10 /
&MESH XB=1,2,0,1,0,1, IJK=10,10,10 /
&TAIL /
```
Run this on both the Windows computer and linux. If this does not work, I would guess that there is something fundamentally wrong with your configuration that I cannot fix remotely.  If it does work, then I suggest you increase the size and number of meshes until a failure occurs. It might be that you do not have enough memory, or you have not set 
```
ulimit -s unlimited
```
to provide an unlimited stacksize.

**Comment by mcgratta** [comment]
Shorten the length of the `ID`s on the `SLCF` lines. They are overflowing the register.

**Reply by ESH-y** [reply]
hi,
do you mean this ID:
&SLCF QUANTITY='PRESSURE', ID='X = 1.0 m: Pressure', PBX=67.0/
&SLCF QUANTITY='VISIBILITY', ID='X = 1.0 m: Visibility', PBX=67.0/
&SLCF QUANTITY='TEMPERATURE', ID='X = 1.0 m: Temperature', PBX=67.0/
&SLCF QUANTITY='VELOCITY', VECTOR=.TRUE., ID='X = 1.0 m: Velocity', PBX=67.0/
&SLCF QUANTITY='VOLUME FRACTION', SPEC_ID='CARBON MONOXIDE', ID='X = 1.0 m: [Species: CARBON MONOXIDE] Volume Fraction', PBX=67.0/

we allways run this way and we never had problem with that before.

**Reply by mcgratta** [reply]
Yes
